src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/GraphEntityDAO.scala

```diff
@@ -39,6 +39,11 @@ class GraphEntityDAO extends EntityDAO with GraphDAO {
   }
 
   private def createVertex(workspace: Vertex, workspaceNamespace: String, workspaceName: String, entity: Entity, txn: RawlsTransaction): Vertex = txn withGraph { db =>
+    // check for illegal dot characters
+    validateUserDefinedString(entity.entityType)
+    validateUserDefinedString(entity.name)
+    entity.attributes.keys.foreach(validateUserDefinedString)
+
```

Comment:
Currently the following things go on edge labels:
1. entity type
2. attribute names
3. static edge labels defined in ```GraphDAO```

So technically it's enough to validate attribute names and entity type, plus making sure our static labels are dotless. But it might be safer to check any time we add an edge label (e.g. add an ```addEdge``` method in ```GraphDAO```, just like we call our own ```addVertex``` instead of using the base method).